feat: Add Harmony release action links to MusicBrainz pages

Dr-Blank opened the pull request:
This userscript adds a convenient link to the Harmony website directly from MusicBrainz pages, making it easier to jump to Harmony's release-specific actions.

Motivation:

Currently, navigating from a MusicBrainz release to its corresponding Harmony actions page requires manually copying the MBID and constructing the URL. This script automates that process by adding a clickable icon.

Functionality:

1. On single release pages, the icon is placed in the main page header next to the release title.

3. On release group and artist releases pages, icons are added next to each release listed in the main table.
 
5.  Clicking any icon opens the corresponding Harmony release actions URL (https://harmony.pulsewidth.org.uk/release/actions?release_mbid=...) in a new browser tab.

Dr-Blank commented:
should i add the built script to the PR?

kellnerd commented:
Hi, my time is limited currently, so I won't get to review this PR before the weekend.

However, I wanted to let you know early that there is already [another userscript](https://github.com/RustyNova016/MusicBrainz-UserScripts/blob/main/musicbrainz-go-to-harmony.user.js) which does a similar task. From a brief look, it seems that your script supports more MB pages, so it might be worth trying to integrate this into the other userscript, if it fits there.

kellnerd reviewed (commented):
I've finally built and tested the userscript, it works fine.

Personally I consider it too invasive to have release actions links for every release on artist and release group pages, the probability that I'm going to use them is probably limited.
Most people would probably go through their favourite artist's releases one time and use the release actions, afterwards they are unlikely to be used again. But maybe I am just not the target audience of this userscript 😀

That said, I would still accept the userscript in this repo if you are willing to maintain it. I can probably fix simple issues (if the website changes, for example), but it's hard for me to guarantee maintainenance of a script which I'm not using myself.

Now some actual feedback about the script itself:

What do you think about injecting the icon link _after_ the release title? It feels a bit cleaner to me to have a secondary action after the title and not next to the entity icon (on release pages) or the cover art indicator icon (in release tables).

I would like to avoid inline style attributes, can you try to style the icon with CSS classes? Ideally making use of one of the existing CSS rules from the website, but if no suitable rules exist, we can always inject our own minimal stylesheet (like [here](https://github.com/kellnerd/musicbrainz-scripts/blob/c8766d6892f6cc7e0ad1ee317fda0c109a856b3c/src/creditParserUI.js#L75)).

Dr-Blank commented:
Thanks for taking the time to build, test, and provide feedback on the userscript! I really appreciate it.

I understand your point about the icons on artist/RG pages potentially being less frequently used after an initial pass. To address it I have added a key value to later turn it off. I'm happy to maintain the script as I use it, also as it might still be useful for users who frequently interact with Harmony or revisit those pages. If maintenance becomes an issue down the line, we can definitely reassess.

Regarding your specific feedback on the script itself:

1.  **Icon Placement:** I've updated the script to inject the Harmony icon *after* the release title/link on all page types (release header, RG table, artist releases table). I agree this feels cleaner as a secondary action.
2.  **Styling:** I've removed all inline style attributes. The icon link and the icon image now use CSS classes (`harmony-userscript-link` and `harmony-userscript-icon`). The necessary styles (including size and margin) are dynamically generated based on configuration and injected into the page's `<head>` via a `<style>` tag.
3.  **Configuration:** The configurable values (icon size and left margin, plus to turn off on RG/Artist pages) are now stored and retrieved using `GM_getValue` and `GM_setValue`. Users can modify these directly in their userscript manager's storage tab if they want to tweak the appearance. Default values are applied if nothing is set.

I've pushed these changes. Let me know what you think!

Thanks again for the review and guidance.

kellnerd commented on src/userscripts/linkHarmonyReleaseActions.user.js:
```diff
@@ -0,0 +1,315 @@
+  harmonyLink.href = `${DEFAULT_HARMONY_BASE_URL}${mbid}`;
+  harmonyLink.target = "_blank";
+  harmonyLink.rel = "noopener noreferrer";
+  harmonyLink.title = `View Harmony Actions for ${mbid} (opens in new tab)`;
```
```suggestion
  harmonyLink.title = `View Harmony Release Actions (opens in new tab)`;
```
I don't think it is necessary to include the MBID in the tooltip, it is not important for the user (and in case it is, the full URL is usually shown in the browser's status bar/tooltip).

kellnerd commented on src/userscripts/linkHarmonyReleaseActions.user.js:
```diff
@@ -0,0 +1,315 @@
+
+    // --- Format Check ---
+    if (digitalOnly) {
+      const formatCell = parentRow.querySelector("td:nth-of-type(4)"); // 4th column is Format
```
This check is a bit unreliable and doesn't work in all scenarios. For example, there is a popular [userscript which adds an additional column](https://github.com/murdos/musicbrainz-userscripts#mb_relationship_shortcuts), shifting the format into the fifth column.
Maybe you could determine the position of the "Format" header cell once for every table and use that index here?

kellnerd commented on src/userscripts/linkHarmonyReleaseActions.user.js:
```diff
@@ -0,0 +1,315 @@
+      }
+    }
+  });
+  // console.log(`Harmony Link Script: Processed table "${tableBodySelector}". Added ${addedCount} icons.`);
```
Instead of commenting out these logger statements, you can use `console.debug`. The bundler will remove them automatically for production builds (`build` task) and only preserve them for debug builds (`debug-build` task).

kellnerd reviewed (changes requested):
Thank you very much for the latest improvements, especially the persistent configuration values. I really like your attention to details and appreciate your patience, sorry that it took me so long to review this again.
Just one last thing remains to be fixed and you have won a new user 😀

Dr-Blank commented:
sorry for the late reply

I have applied your suggestions as well as added logic to get format column from instead of hard coding the value.

It works with the userscript you linked. However it worked without the patch as well! I think it must be because my script ran before the other one could add a new column, and hence i could not test my scripts failure before this patch. Regardless, it should now work with the new patch with any number of columns being added/removed.
